app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-specialist-link-tracker.js

```diff
@@ -0,0 +1,200 @@
+
+        document.querySelector('body').addEventListener('click', this.handleClick)
+        document.querySelector('body').addEventListener('contextmenu', this.handleClick)
+        document.querySelector('body').addEventListener('mousedown', this.handleMousedown)
+      }
+    },
+
+    stopTracking: function () {
+      document.querySelector('body').removeEventListener('click', this.handleClick)
+      document.querySelector('body').removeEventListener('contextmenu', this.handleClick)
+      document.querySelector('body').removeEventListener('mousedown', this.handleMousedown)
+    },
+
+    handleClick: function (event) {
+      var element = event.target
+
+      if (element.tagName !== 'A') {
+        element = element.closest('a')
+      }
+
+      if (!element) {
+        return
+      }
+
+      var href = element.getAttribute('href')
+
+      if (!href) {
+        return
+      }
+      var clickData = {}
+      var linkAttributes = element.getAttribute('data-ga4-link')
+      if (linkAttributes) {
+        clickData = JSON.parse(linkAttributes)
+
+        /* Since external links can't be determined in the template, we use populated-via-js as a signal
+        for our JavaScript to determine this value. */
+        if (clickData.external === 'populated-via-js' && clickData.url) {
+          clickData.external = window.GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(clickData.url, this.internalDomains) ? 'true' : 'false'
+        }
+
+        if (clickData.method === 'populated-via-js') {
+          clickData.method = window.GOVUK.analyticsGa4.core.trackFunctions.getClickType(event)
+        }
+
+        if (clickData.index) {
+          clickData.index = parseInt(clickData.index)
+        }
+
+        if (clickData.index_total) {
+          clickData.index_total = parseInt(clickData.index_total)
+        }
+      } else if (window.GOVUK.analyticsGa4.core.trackFunctions.isMailToLink(href)) {
+        clickData.event_name = 'navigation'
+        clickData.type = 'email'
+        clickData.external = 'true'
+        clickData.url = href
+      } else if (this.isDownloadLink(href)) {
+        clickData.event_name = 'file_download'
+        clickData.type = this.isPreviewLink(href) ? 'preview' : 'generic download'
+        clickData.external = window.GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(href, this.internalDomains) ? 'true' : 'false'
+        clickData.url = href
+      } else if (window.GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(href, this.internalDomains)) {
+        clickData.event_name = 'navigation'
+        clickData.type = 'generic link'
+        clickData.external = 'true'
+        clickData.url = href
+      }
+
+      if (Object.keys(clickData).length > 0) {
+        if (clickData.url) {
+          clickData.url = window.GOVUK.analyticsGa4.core.trackFunctions.removeCrossDomainParams(clickData.url)
+          clickData.link_domain = this.populateLinkDomain(clickData.url)
```

Comment:
This needs to be implemented across both trackers (Sorry, I thought I mentioned it in my last review but I can't seem to find my comment so I must have forgot to comment it). This is what the analysts use to reconstruct the URL with the `link_path_parts`.

Reply:
Thanks, have now added.